diff --git a/third_party/tf_runtime/workspace.bzl b/third_party/tf_runtime/workspace.bzl
--- a/third_party/tf_runtime/workspace.bzl
+++ b/third_party/tf_runtime/workspace.bzl
@@ -25,4 +25,0 @@
-        repo_mapping = {
-            "@tsl": "@tsl",
-            "@xla": "@xla",
-        },
diff --git a/third_party/xla/xla/tsl/cuda/BUILD.bazel b/third_party/xla/xla/tsl/cuda/BUILD.bazel
--- a/third_party/xla/xla/tsl/cuda/BUILD.bazel
+++ b/third_party/xla/xla/tsl/cuda/BUILD.bazel
@@ -47,7 +47,7 @@ cc_library(
         "//xla/tsl/platform:logging",
         "@com_google_absl//absl/container:flat_hash_set",
         "@local_config_cuda//cuda:cuda_headers",
-        "@tsl//tsl/platform:dso_loader",
+        "//xla/tsl/platform/default:dso_loader",
         "@tsl//tsl/platform:load_library",
     ]),
 )
@@ -78,7 +78,7 @@ cc_library(
         # keep sorted
         "//xla/tsl/platform:logging",
         "@local_config_cuda//cuda:cuda_headers",
-        "@tsl//tsl/platform:dso_loader",
+        "//xla/tsl/platform/default:dso_loader",
         "@tsl//tsl/platform:load_library",
     ]),
 )
@@ -109,7 +109,7 @@ cc_library(
         # keep sorted
         "//xla/tsl/platform:logging",
         "@local_config_cuda//cuda:cuda_headers",
-        "@tsl//tsl/platform:dso_loader",
+        "//xla/tsl/platform/default:dso_loader",
         "@tsl//tsl/platform:load_library",
     ]),
 )
@@ -146,7 +146,7 @@ cc_library(
             "//xla/tsl/platform:logging",
             "@com_google_absl//absl/container:flat_hash_set",
             "@local_config_cuda//cuda:cuda_headers",
-            "@tsl//tsl/platform:dso_loader",
+            "//xla/tsl/platform/default:dso_loader",
             "@tsl//tsl/platform:load_library",
         ],
         "//conditions:default": [],
@@ -181,7 +181,7 @@ cc_library(
         "//xla/tsl/platform:logging",
         "@com_google_absl//absl/container:flat_hash_map",
         "@local_config_cuda//cuda:cudnn_header",
-        "@tsl//tsl/platform:dso_loader",
+        "//xla/tsl/platform/default:dso_loader",
         "@tsl//tsl/platform:load_library",
     ]),
 )
@@ -235,7 +235,7 @@ cc_library(
         # keep sorted
         "//xla/tsl/platform:logging",
         "@local_config_cuda//cuda:cuda_headers",
-        "@tsl//tsl/platform:dso_loader",
+        "//xla/tsl/platform/default:dso_loader",
         "@tsl//tsl/platform:load_library",
     ]),
 )
@@ -273,7 +273,7 @@ cc_library(
         "//xla/tsl/platform:logging",
         "@local_config_cuda//cuda:cuda_headers",
         "@local_config_cuda//cuda:cupti_headers",
-        "@tsl//tsl/platform:dso_loader",
+        "//xla/tsl/platform/default:dso_loader",
         "@tsl//tsl/platform:load_library",
     ]),
 )
@@ -309,7 +309,7 @@ cc_library(
         # keep sorted
         "//xla/tsl/platform:logging",
         "@local_config_cuda//cuda:cuda_headers",
-        "@tsl//tsl/platform:dso_loader",
+        "//xla/tsl/platform/default:dso_loader",
         "@tsl//tsl/platform:load_library",
     ]),
 )
@@ -346,7 +346,7 @@ cc_library(
         "//xla/tsl/platform:logging",
         "@com_google_absl//absl/container:flat_hash_set",
         "@local_config_cuda//cuda:cuda_headers",
-        "@tsl//tsl/platform:dso_loader",
+        "//xla/tsl/platform/default:dso_loader",
         "@tsl//tsl/platform:load_library",
     ]),
 )
@@ -381,7 +381,7 @@ cc_library(
         "@com_google_absl//absl/strings:string_view",
         "@local_config_cuda//cuda:cuda_headers",
         "@local_config_nccl//:nccl_headers",
-        "@tsl//tsl/platform:dso_loader",
+        "//xla/tsl/platform/default:dso_loader",
         "@tsl//tsl/platform:load_library",
     ]),
 )
@@ -413,7 +413,7 @@ cc_library(
     deps = if_cuda_is_configured([
         # keep sorted
         "//xla/tsl/platform:logging",
-        "@tsl//tsl/platform:dso_loader",
+        "//xla/tsl/platform/default:dso_loader",
         "@tsl//tsl/platform:load_library",
     ]),
 )
@@ -444,7 +444,7 @@ cc_library(
         # keep sorted
         "//xla/tsl/platform:logging",
         "@local_config_cuda//cuda:cuda_headers",
-        "@tsl//tsl/platform:dso_loader",
+        "//xla/tsl/platform/default:dso_loader",
         "@tsl//tsl/platform:load_library",
     ]),
 )
@@ -475,7 +475,7 @@ cc_library(
         # keep sorted
         "//xla/tsl/platform:logging",
         "@local_config_cuda//cuda:cuda_headers",
-        "@tsl//tsl/platform:dso_loader",
+        "//xla/tsl/platform/default:dso_loader",
         "@tsl//tsl/platform:load_library",
     ]),
 )
