diff -r -u old/tensorflow/c/BUILD new/tensorflow/c/BUILD
--- old/tensorflow/c/BUILD	2020-02-06 13:20:27.000000000 -0800
+++ new/tensorflow/c/BUILD	2020-03-06 16:02:29.239566981 -0800
@@ -694,4 +694,5 @@
         # TODO(b/74620627): remove when _USE_C_SHAPES is removed
         "//tensorflow/python:cpp_shape_inference_proto_cc",
     ],
+    alwayslink = 1,
 )
diff -r -u old/tensorflow/core/grappler/clusters/BUILD new/tensorflow/core/grappler/clusters/BUILD
--- old/tensorflow/core/grappler/clusters/BUILD	2020-02-06 13:20:27.000000000 -0800
+++ new/tensorflow/core/grappler/clusters/BUILD	2020-03-06 16:47:11.057163837 -0800
@@ -138,6 +138,7 @@
         "//tensorflow/core/grappler:utils",
         "//tensorflow/core/kernels:ops_util",
     ],
+    alwayslink = 1,
 )
 
 tf_cc_test(
diff -r -u old/tensorflow/core/grappler/costs/BUILD new/tensorflow/core/grappler/costs/BUILD
--- old/tensorflow/core/grappler/costs/BUILD	2020-02-06 13:20:27.000000000 -0800
+++ new/tensorflow/core/grappler/costs/BUILD	2020-03-06 16:45:14.316942148 -0800
@@ -303,6 +303,7 @@
         "//tensorflow/core/grappler/costs:cost_estimator",
         "//tensorflow/core/kernels:ops_util",
     ],
+    alwayslink = 1,
 )
 
 cc_library(
@@ -319,6 +320,7 @@
         "//tensorflow/core:protos_all_cc",
         "//tensorflow/core/grappler/clusters:utils",
     ] + tf_protos_grappler(),
+    alwayslink = 1,
 )
 
 tf_cc_test(
@@ -353,6 +355,7 @@
         "//tensorflow/core/grappler:op_types",
         "//tensorflow/core/grappler:utils",
     ] + tf_protos_grappler(),
+    alwayslink = 1,
 )
 
 tf_cc_test(
diff -r -u old/tensorflow/core/platform/BUILD new/tensorflow/core/platform/BUILD
--- old/tensorflow/core/platform/BUILD	2020-02-06 13:20:27.000000000 -0800
+++ new/tensorflow/core/platform/BUILD	2020-03-06 16:00:30.423279302 -0800
@@ -578,6 +578,7 @@
     name = "stacktrace_handler",
     textual_hdrs = ["stacktrace_handler.h"],
     deps = tf_windows_aware_platform_deps("stacktrace_handler"),
+    alwayslink = 1,
 )
 
 cc_library(
diff -r -u old/tensorflow/core/platform/default/BUILD new/tensorflow/core/platform/default/BUILD
--- old/tensorflow/core/platform/default/BUILD	2020-02-06 13:20:27.000000000 -0800
+++ new/tensorflow/core/platform/default/BUILD	2020-03-06 16:00:44.203312903 -0800
@@ -350,6 +350,7 @@
         "//tensorflow/core/platform",
         "//tensorflow/core/platform:stacktrace",
     ],
+    alwayslink = 1,
 )
 
 cc_library(
diff -r -u old/tensorflow/python/BUILD new/tensorflow/python/BUILD
--- old/tensorflow/python/BUILD	2020-02-06 13:20:27.000000000 -0800
+++ new/tensorflow/python/BUILD	2020-03-06 16:43:09.652704387 -0800
@@ -496,6 +496,7 @@
         "//tensorflow/core:protos_all_cc",
         "//third_party/python_runtime:headers",
     ],
+    alwayslink = 1,
 )
 
 cc_library(
@@ -5650,6 +5651,7 @@
         "//third_party/py/numpy:headers",
         "//third_party/python_runtime:headers",
     ],
+    alwayslink = 1,
 )
 
 py_library(
